src/dataLoaders/components/collectorsWizard/select/StreamingSelector.tsx

```diff
@@ -71,52 +71,64 @@ class StreamingSelector extends PureComponent<Props, State> {
-                onChange={this.handleFilterChange}
-                placeholder="Filter Plugins..."
-              />
-            </FormElement>
-          </Grid.Column>
-        </Grid.Row>
-        <SquareGrid cardSize={cardSize} gutter={ComponentSize.Small}>
-          {this.filteredBundles.map(b => {
-            return (
-              <SquareGrid.Card key={b}>
-                <SelectableCard
-                  id={b}
-                  formName="telegraf-plugins"
-                  label={b}
-                  testID={`telegraf-plugins--${b}`}
-                  selected={this.isCardChecked(b)}
-                  onClick={this.handleToggle}
-                  icon={IconFont.Checkmark}
-                >
-                  {createElement(BUNDLE_LOGOS[b])}
-                </SelectableCard>
-              </SquareGrid.Card>
-            )
-          })}
-        </SquareGrid>
+        {buckets.length ? (
+          <>
+            <Grid.Row>
+              <Grid.Column widthSM={Columns.Five}>
+                <FormElement label="Bucket">
+                  <BucketDropdown
+                    selectedBucketID={this.selectedBucketID}
+                    buckets={buckets}
+                    onSelectBucket={this.handleSelectBucket}
+                  />
+                </FormElement>
+              </Grid.Column>
+              <Grid.Column widthSM={Columns.Five} offsetSM={Columns.Two}>
+                <FormElement label="">
+                  <Input
+                    className="wizard-step--filter"
+                    size={ComponentSize.Small}
+                    icon={IconFont.Search}
+                    value={searchTerm}
+                    onBlur={this.handleFilterBlur}
+                    onChange={this.handleFilterChange}
+                    placeholder="Filter Plugins..."
+                  />
+                </FormElement>
+              </Grid.Column>
+            </Grid.Row>
+            <SquareGrid cardSize={cardSize} gutter={ComponentSize.Small}>
+              {this.filteredBundles.map(b => {
+                return (
+                  <SquareGrid.Card key={b}>
+                    <SelectableCard
+                      id={b}
+                      formName="telegraf-plugins"
+                      label={b}
+                      testID={`telegraf-plugins--${b}`}
+                      selected={this.isCardChecked(b)}
+                      onClick={this.handleToggle}
+                      icon={IconFont.Checkmark}
+                    >
+                      {createElement(BUNDLE_LOGOS[b])}
+                    </SelectableCard>
+                  </SquareGrid.Card>
+                )
+              })}
+            </SquareGrid>
+          </>
+        ) : (
+          <>
+            <Grid.Row style={{width: 'auto'}}>
+              <CreateBucketButton />
+            </Grid.Row>
+            <h3 data-testid="create-bucket-prompt">
+              Create a Bucket To Show Telegraf Plugins
```

Comment:
🤔 What if we changed this to "You can't install a telegraf plugin until you create a bucket"

I'm not sure this current verbiage explains why the user would want to show the telegraf plugins and it feels a little gatekeeping - like "you have to make a bucket before we show you this page."

I'd prefer to explain a bit of the process to the user as we solve their problem - basically some variation of "without a bucket, you can't write to anywhere, so you need a bucket" what do you think? also @daviddkkim and @russorat

Reply:
Yeah this was just the first thing that popped into my head, fifgured i would throw it in and discuss it when it got pushed up, thanks for bringing this up here